src/metadata_constraints.cpp

```diff
@@ -30,3 +30,27 @@ METADATA_ASSERT("LongDescription shouldn't be shorter than Description")
   return !data.has("LongDescription")
       || data["LongDescription"].size() >= data["Description"].size();
 }
+
+METADATA_ASSERT("Illustration must exist and be a 48x48 PNG file")
+{
+  if (!data.has("Illustration_48x48@1")) {
+    return false;
+  }
+
+  const std::string& illustration = data["Illustration_48x48@1"];
+
+  if (illustration.length() >= 24) {
```

Comment:
Why is this test needed? What's wrong with always (unconditionally) testing the first 8 bytes?